src/omv/imlib/simd.h

```diff
@@ -0,0 +1,1455 @@
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vhaddq(v0.u16, v1.u16);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __UHADD16(v0.u32[0], v1.u32[0]) }
+    };
+    #else
+    return (v128_t) {
+               .u16 = (v0.u16 + v1.u16) >> 1
+    };
+    #endif
+}
+
+static inline v128_t vhadd_s16(v128_t v0, v128_t v1) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vhaddq(v0.s16, v1.s16);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .s32 = { __SHADD16(v0.s32[0], v1.s32[0]) }
+    };
+    #else
+    return (v128_t) {
+               .s16 = (v0.s16 + v1.s16) >> 1
+    };
+    #endif
+}
+
+static inline v128_t vuxtb16(v128_t v0) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vmovlbq(v0.u8);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __UXTB16(v0.u32[0]) }
+    };
+    #else
+    v128_t r;
+    r.u16[0] = v0.u8[0];
+    r.u16[1] = v0.u8[2];
+    return r;
+    #endif
+}
+
+static inline v128_t vsxtb16(v128_t v0) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vmovlbq(v0.s8);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __SXTB16(v0.u32[0]) }
+    };
+    #else
+    v128_t r;
+    r.s16[0] = v0.s8[0];
+    r.s16[1] = v0.s8[2];
+    return r;
+    #endif
+}
+
+static inline v128_t vuxtb16_ror8(v128_t v0) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vmovltq(v0.u8);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __UXTB16_RORn(v0.u32[0], 8) }
+    };
+    #else
+    v128_t r;
+    r.u16[0] = v0.u8[1];
+    r.u16[1] = v0.u8[3];
+    return r;
+    #endif
+}
+
+static inline v128_t vsxtb16_ror8(v128_t v0) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vmovltq(v0.s8);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __UXTB16_RORn(v0.u32[0], 8) }
+    };
+    #else
+    v128_t r;
+    r.s16[0] = v0.s8[1];
+    r.s16[1] = v0.s8[3];
+    return r;
+    #endif
+}
+
+static inline v128_t vpkhbt(v128_t v0, v128_t v1) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vsliq_n_u32(v0.u32, v1.u32, 16);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __PKHBT(v0.u32[0], v1.u32[0], 16) }
+    };
+    #else
+    v128_t r;
+    r.u16[0] = v0.u16[0];
+    r.u16[1] = v1.u16[0];
+    return r;
+    #endif
+}
+
+static inline v128_t vpkhbt_ror8(v128_t v0, v128_t v1) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vshrntq_n_u32(v0.u16, v1.u32, 8);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __PKHBT(v0.u32[0], v1.u32[0], 8) }
+    };
+    #else
+    v128_t r;
+    r.u16[0] = v0.u16[0];
+    r.u16[1] = v1.u32[0] >> 8;
+    return r;
+    #endif
+}
+
+static inline v128_t vpkhtb(v128_t v0, v128_t v1) {
+    #if (__ARM_ARCH >= 8)
+    return (v128_t) vsriq_n_u32(v0.u32, v1.u32, 16);
+    #elif (__ARM_ARCH >= 7)
+    return (v128_t) {
+               .u32 = { __PKHTB(v0.u32[0], v1.u32[0], 16) }
```

Comment:
```suggestion
        .u32 = { __PKHTB(v0.u32[0], v1.u32[0], 16) }
```
Seems like there's an extra 4 spaces in all these lines.

Reply:
Formatter requires that. If I remove it then it won't pass.

Reply:
This case is probably not handled correctly in the config, will see if I can fix it.

Reply:
I fixed this in the config, if you rebase and reformat it will use 4 spaces.

Reply:
Updated... checking if passing now.